### ISSUE TITLE
[ban-types] False positive when importing symbol with name Function

### ISSUE THREAD

**Issue by albshin** [typescript-eslint/typescript-eslint#4794]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-types": "error"
  }
}
```

```TS
import { Function } from '@aws-cdk/aws-lambda';

export interface LambdaStack {
  readonly handlerLambda: Function;
}
```

**Expected Result**

The rule detects that the imported Function is not global Function (either from the import or the parameters).

**Actual Result**

The following error is given:

```
Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape.eslint[@typescript-eslint/ban-types](https://typescript-eslint.io/rules/ban-types)
```

**Additional Info**

Same issue as https://github.com/typescript-eslint/typescript-eslint/issues/2338 but for `ban-types`.
I am using an older version of CDK (v1) but CDK v2 also exports a `Function` type.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.16.0` |
| `@typescript-eslint/parser`        | `5.16.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `8.12.0` |
| `node`                             | `12.22.1` |

**Comment by bradzacher** [comment]
The rule does not differentiate between global or imported types by design.

It simply bans a type name, period.

I would suggest aliasing the import here because shadowing the global name is a very good way to confuse people.
